common/core/devdoc/message_requirements.md

```diff
@@ -0,0 +1,55 @@
+# azure-iot-common.Message Requirements
+========================================
+
+## Overview
+azure-iot-common.Message provides a container for message content and properties travelling to and from an IoT Hub.
+
+## Example usage
+```
+'use strict';
+var Message = require('azure-iot-common').Message;
+
+var message = new Message('data to send');
+message.properties.add('myProperty', 'my property value');
+console.log('message content: ' + message.getData());
+```
+
+## Public Interface
+
+| **Name**              | **Type**                | **Description** |
+|-----------------------|-------------------------|-----------------|
+| message.properties    | Object<string, string>  | A container of custom message properties |
+| message.messageId     | String                  |  Used to correlate two-way communication. Format: A case-sensitive string ( up to 128 char long) of ASCII 7-bit alphanumeric chars + {'-', ':',’.', '+', '%', '_', '#', '*', '?', '!', '(', ')', ',', '=', '@', ';', '$', '''}. |
+| message.to            |                         | Destination of the message |
+| message.expiryTime    | Date                    | Expiry time in UTC. Interpreted by hub on C2D messages. Ignored in other cases |
+| message.lockToken     | String                  | Used by receiver to Abandon, Reject or Complete the message |
+| message.correlationId | String                  | Used in message responses and feedback |
+| message.userId        | String                  | Used to specify the entity creating the message |
```

Comment:
>userId [](start = 10, length = 6)

missing contentType and encoding

Reply:
oh it's just copy/paste of the old docm file. never mind then - it's unrelated to your changes

---
In reply to: [181909799](https://github.com/Azure/azure-iot-sdk-node/pull/271#discussion_r181909799) [](ancestors = 181909799)